modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -1033,6 +1034,10 @@ public GranuleLoadingResult loadRaster(final ImageReadParam imageReadParameters,
             // image sizes.
             //
             // place it in the mosaic using the coords created above;
+            if (virtualNativeResolution != null && !Double.isNaN(virtualNativeResolution[0]) && !Double.isNaN(virtualNativeResolution[1])){
```

Comment:
Where do you check that the requested resolution is higher than the virtual one? Like, if we are reading an overview that has lower resolution than the virtual one, no rescaling should be applied, right?